### PR TITLE
Validate batch merkle root

### PR DESCRIPTION
**Pull request by mooselumph**
## Why are these changes needed?

Why is it needed, indeed. 

## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: core/validator.go [hunk 1]
```diff
@@ -13,30 +13,29 @@ var (
 	ErrBlobQuorumSkip      = errors.New("blob skipped for a quorum before verification")
 )
 
-type ChunkValidator interface {
-	ValidateBatch([]*BlobMessage, *OperatorState, common.WorkerPool) error
-	ValidateBlob(*BlobMessage, *OperatorState) error
+type DataValidator interface {
```
Comment (jianoaix) [inline core/validator.go:16]: shall we create a name for data that's received by node (e.g. shard, BlobShard)? This naming may be more specific

**Review by jianoaix — APPROVED** [review]

File: core/validator.go [hunk 2]
```diff
@@ -226,3 +201,27 @@ func (v *chunkValidator) VerifyBlobLengthWorker(blobCommitments encoding.BlobCom
 
 	out <- nil
 }
+
+func ValidateBatchHeaderRoot(batchHeader *BatchHeader, blobs []*BlobMessage) error {
```
Comment (ian-shim) [inline core/validator.go:205]: nit: make it private?

**Review by ian-shim — APPROVED** [review]

File: core/validator.go (hunk 1 above)
Comment (mooselumph) [reply core/validator.go:16]: Are you suggesting changing `BlobMessage` into `BlobShard`?

Comment (jianoaix) [reply core/validator.go:16]: I'm suggesting "DataValidator" to something more specific such as "BlobShardValidator"

Comment (bxue-l2) [reply core/validator.go:16]: DataMessageValidator?

**Review by bxue-l2 — APPROVED** [review]

File: core/validator.go (hunk 1 above)
Comment (mooselumph) [reply core/validator.go:16]: What about just `ShardValidator`?

**Pull request merged**